File: flutter_app/lib/utils/soft_break_korean_math_text.dart
/// 웹 `src/lib/soft-break-korean-math-text.ts` 와 동일 규칙
String softBreakAnswerExplanation(String input) {
  var s = input.replaceAll('\r\n', '\n');
  if (s.trim().isEmpty) return s;
  if (s.contains('\n')) return s;

  final expl = RegExp(
    r'\s+(?=코사인|사인\s*법칙|법칙을\s*이용|따라서|그러므로|먼저|여기(?:서|부터)|이를\s*이용|식(?:으로|을)\s*|두\s*삼각형|넓이|반지름|각도\s*를|평균|삼각(?:형|함수))',
  );
  final m = expl.firstMatch(s);
  if (m == null) return s;

  final head = s.substring(0, m.start).trimRight();
  final tail = s.substring(m.start).trimLeft();
  return '$head\n\n$tail';
}

File: src/lib/soft-break-korean-math-text.ts
/**
 * 모델이 한 줄로만 붙여 보낸 "정답 문장 + 풀이 설명"을 읽기 좋게 나눕니다.
 * 이미 줄바꿈이 있으면 건드리지 않습니다.
 */
export function softBreakAnswerExplanation(input: string): string {
  const s = input.replace(/\r\n/g, "\n");
  if (!s.trim()) {
    return s;
  }
  if (/\n/.test(s)) {
    return s;
  }

  const expl = /\s+(?=코사인|사인\s*법칙|법칙을\s*이용|따라서|그러므로|먼저|여기(?:서|부터)|이를\s*이용|식(?:으로|을)\s*|두\s*삼각형|넓이|반지름|각도\s*를|평균|삼각(?:형|함수))/;
  const m = expl.exec(s);
  if (!m || m.index === undefined) {
    return s;
  }

  const head = s.slice(0, m.index).trimEnd();
  const tail = s.slice(m.index).trimStart();
  return `${head}\n\n${tail}`;
}
